Error on signup with taken username

lambdaTotoro opened the pull request:
Fixes #91, including an additional test for signing up with the same username multiple times and a fitting error message on rejected signup.

In the course of this, I refactored the `get_or_create_user()` method into three methods: `user_exists()`, `get_user()` and `create_user()` which gets rid of the signup error (which was related to getting a user when, really, you wanted to create a user).

I also replaced calls to UserInfo without username normalization in the authenticator with calls to the aforementioned `get_user()`, which does include username normalization.

leportella commented on nativeauthenticator/nativeauthenticator.py:
```diff
@@ -166,13 +166,22 @@ def is_password_strong(self, password):
-        user = UserInfo.find(self.db, username)
+        
+    def get_user(self, username):
+        user = UserInfo.find(self.db, self.normalize_username(username))
```
If I am not mistaken, if the user doesn't exist it already returns None, so we can remove the following if and just return the result of this search

leportella commented on nativeauthenticator/nativeauthenticator.py:
```diff
@@ -166,13 +166,22 @@ def is_password_strong(self, password):
+    def get_user(self, username):
+        user = UserInfo.find(self.db, self.normalize_username(username))
         if user:
             return user
```
then this won't be necessary

leportella commented on nativeauthenticator/nativeauthenticator.py:
```diff
@@ -166,13 +166,22 @@ def is_password_strong(self, password):
+        return self.get_user(username) is not None
 
+    def create_user(self, username, pw, **kwargs):
+        
```
can you remove this line to follow patterns please?

leportella commented:
I asked for a couple of minor things but it looks great! Thanks for your pull request

lambdaTotoro commented:
I have incorporated the changes you pointed out. You were absolutely correct, the `if user:`-block was entirely unnecessary. I also added a test for this method.

leportella commented:
thanks!
